android/src/main/kotlin/com/mapbox/maps/mapbox_maps/MapboxMapController.kt

```diff
@@ -104,19 +106,18 @@ class MapboxMapController(
     }
 
     override fun onDestroy(owner: LifecycleOwner) {
-      lifecycleRegistry.currentState = Lifecycle.State.DESTROYED
+      lifecycleRegistry.currentState = when (shouldDestroyOnDestroy) {
+        true -> Lifecycle.State.DESTROYED
+        false -> Lifecycle.State.CREATED
+      }
     }
 
     fun dispose() {
       parentLifecycle.removeObserver(this)
-      // fires MapView.onStop
-      lifecycleRegistry.currentState = Lifecycle.State.CREATED
-      // fires MapView.onDestroy
-      lifecycleRegistry.currentState = Lifecycle.State.DESTROYED
     }
```

Comment:
Hmm, so now when dispose is called in `onFlutterViewDetached` - lifecycle registry is not reset to CREATED state at all? How does this work, `onFlutterViewDetached` is always called after `onDestroy` ?

Reply:
Good point, I couldn't find a confirmation that disposal would always be preceded by onDestroy, Added it back.